modules/models/src/main/java/org/eclipse/xpanse/modules/models/admin/IdentityProviderStatus.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+package org.eclipse.xpanse.modules.models.admin;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+import org.eclipse.xpanse.modules.models.admin.enums.HealthStatus;
+import org.eclipse.xpanse.modules.models.admin.enums.IdentityProviderType;
+
+/**
+ * Describes health status of the identity provider.
+ */
+@Data
+public class IdentityProviderStatus {
```

Comment:
Even here there is a small change in the requirement. 
Instead of separate type for each 'integrated' component, we need a common type that can fit all types of systems. 

BackendSystem ->
Name(string) - value should be 'zitadel' if Zitadel is enabled, 
Endpoint(string)
SystemStatus
SystemTypeEnum -> current values - IdentityProvider, Database

the HealthCheck object must contain a List<BackendSystem>

sorry for the confusion.

Reply:
a List ? which object should be a list? the error message?

Reply:
list of backend system. We just need one generic type which covers identity provider and database

Reply:
Ok.